Unfolding/20240328_Unfolding/matchingEffCorr.cpp

```diff
@@ -0,0 +1,787 @@
+// ./matchingEffCorr.exe --Input v0/LEP1MC1994_recons_aftercut-001_Matched.root --Matched PairTree --Unmatched UnmatchedPairTree
+#include <iostream>
+#include <vector>
+#include <map>
+#include <filesystem>
+using namespace std;
+
+// root includes
+#include "TTree.h"
+#include "TChain.h"
+#include "TFile.h"
+#include "TTreeReader.h"
+#include "TTreeReaderValue.h"
+#include "TTreeReaderArray.h"
+#include "TStyle.h"
+#include "TH2D.h"
+#include "TCanvas.h"
+#include "TGraph.h"
+#include "TGaxis.h"
+#include "TLatex.h"
+#include "TLegend.h"
+
+#include "Messenger.h"
+#include "CommandLine.h"
+#include "Matching.h"
+#include "ProgressBar.h"
+#include "TauHelperFunctions3.h"
+#include "SetStyle.h"
+#include "EffCorrFactor.h"
+
+int main(int argc, char *argv[]);
+int FindBin(double Value, int NBins, double Bins[]); 
+void MakeCanvasZ(vector<TH1D > Histograms, vector<string> Labels, string Output, string X, string Y, double WorldMin, double WorldMax, bool DoRatio, bool LogX); 
```

Comment:
Same comment as in the other code about `vector<TH1D>` types

Reply:
done